components/style/values/computed/mod.rs

```diff
@@ -494,6 +495,24 @@ impl SVGPaint {
 /// <length> | <percentage> | <number>
 pub type LengthOrPercentageOrNumber = Either<Number, LengthOrPercentage>;
 
+#[cfg(feature = "gecko")]
+impl ToAnimatedValue for LengthOrPercentageOrNumber {
```

Comment:
That doesn't strike me as something that should implement `ToAnimatedValue`. Probably where you were using `LengthOrPercentageOrNumber` you should use a different type instead that conveys that numbers are just unit-less pixel lengths.